homeassistant/components/fritz/config_flow.py

```diff
@@ -0,0 +1,248 @@
+
+        errors = {}
+
+        self._host = user_input[CONF_HOST]
+        self._port = user_input[CONF_PORT]
+        self._username = user_input[CONF_USERNAME]
+        self._password = user_input[CONF_PASSWORD]
+
+        error = await self.fritz_tools_init()
+
+        if not error:
+            self._name = self.fritz_tools.device_info["model"]
+
+            for entry in self._async_current_entries(include_ignore=False):
+                if entry.data[CONF_HOST] == self._host:
+                    error = "already_configured"
+
+        if error:
+            errors["base"] = error
+            return self._show_setup_form_init(errors)
+
+        return self._async_create_entry()
+
+    async def async_step_reauth(self, data):
+        """Handle flow upon an API authentication error."""
+        self._entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        self._host = data[CONF_HOST]
+        self._port = data[CONF_PORT]
+        self._username = data[CONF_USERNAME]
+        self._password = data[CONF_PASSWORD]
+        return await self.async_step_reauth_confirm()
+
+    def _show_setup_form_reauth_confirm(self, user_input, errors=None):
+        """Show the reauth form to the user."""
+        return self.async_show_form(
+            step_id="reauth_confirm",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_USERNAME, default=user_input.get(CONF_USERNAME)
+                    ): str,
+                    vol.Required(CONF_PASSWORD): str,
+                }
+            ),
+            description_placeholders={"host": self._host},
+            errors=errors or {},
+        )
+
+    async def async_step_reauth_confirm(self, user_input=None):
+        """Dialog that informs the user that reauth is required."""
+        if user_input is None:
+            return self._show_setup_form_reauth_confirm(
+                user_input={CONF_USERNAME: self._username}
+            )
+
+        errors = {}
+
+        self._username = user_input[CONF_USERNAME]
+        self._password = user_input[CONF_PASSWORD]
+
+        error = await self.fritz_tools_init()
+
+        if error:
+            errors["base"] = error
+            return self._show_setup_form_reauth_confirm(
+                user_input=user_input, errors=errors
+            )
+
+        self.hass.config_entries.async_update_entry(
+            self._entry,
+            data={
+                CONF_HOST: self._host,
+                CONF_PASSWORD: self._password,
+                CONF_PORT: self._port,
+                CONF_USERNAME: self._username,
+            },
+        )
+        await self.hass.config_entries.async_reload(self._entry.entry_id)
+        return self.async_abort(reason="reauth_successful")
+
+    async def async_step_import(self, import_config):
+        """Import a config entry from configuration.yaml."""
+        return await self.async_step_user(import_config)
```

Comment:
This won't work since `async_step_user` is currently passing right to `return await self.async_step_start_config()` without the import_config